RecoTracker/MkFitCore/interface/TrackerInfo.h

```diff
@@ -0,0 +1,245 @@
+#ifndef RecoTracker_MkFitCore_interface_TrackerInfo_h
+#define RecoTracker_MkFitCore_interface_TrackerInfo_h
+
+#include <string>
+#include <vector>
+#include <stdexcept>
+
+#include <cmath>
+#include <cstdio>
+
+namespace mkfit {
+
+  class IterationsInfo;
+
+  //==============================================================================
+
+  enum WithinSensitiveRegion_e { WSR_Undef = -1, WSR_Inside = 0, WSR_Edge, WSR_Outside };
+
+  struct WSR_Result {
+    // Could also store XHitSize count equivalent here : 16;
+    WithinSensitiveRegion_e m_wsr : 8;
+    bool m_in_gap : 8;
+
+    WSR_Result() : m_wsr(WSR_Undef), m_in_gap(false) {}
+
+    WSR_Result(WithinSensitiveRegion_e wsr, bool in_gap) : m_wsr(wsr), m_in_gap(in_gap) {}
+  };
+
+  //==============================================================================
+
+  class LayerInfo {
+  private:
+    bool is_in_r_hole_no_check(float r) const { return r > m_hole_r_min && r < m_hole_r_max; }
+
+  public:
+    enum LayerType_e { Undef = -1, Barrel = 0, EndCapPos = 1, EndCapNeg = 2 };
+
+    int m_layer_id = -1;
+    LayerType_e m_layer_type = Undef;
+
+    float m_rin, m_rout, m_zmin, m_zmax;
+    float m_propagate_to;
+
+    int m_next_barrel = -1, m_next_ecap_pos = -1, m_next_ecap_neg = -1;
+    int m_sibl_barrel = -1, m_sibl_ecap_pos = -1, m_sibl_ecap_neg = -1;
+
+    bool m_is_outer = false;
+    bool m_has_r_range_hole = false;
+    float m_hole_r_min, m_hole_r_max;  // This could be turned into std::function when needed.
+
+    /*MM: moving out to IterationLayerConfig*/
+    //// Selection limits
+    float m_q_bin;  // > 0 - bin width, < 0 - number of bins
+    //float         m_select_min_dphi, m_select_max_dphi;
+    //float         m_select_min_dq,   m_select_max_dq;
```

Comment:
remove

Reply:
Addressed in Round 2 follow-up PR trackreco/cmssw#76 commit https://github.com/trackreco/cmssw/pull/76/commits/33131c1d2444a1b33880f64ceef661df100dece1